core-service/src/main/java/com/crmfoodestablishment/coreservice/service/MenuService.java

```diff
@@ -0,0 +1,72 @@
+package com.crmfoodestablishment.coreservice.service;
+
+import com.crmfoodestablishment.coreservice.dto.MenuDto;
+import com.crmfoodestablishment.coreservice.entity.Menu;
+import com.crmfoodestablishment.coreservice.mapper.MenuMapper;
+import com.crmfoodestablishment.coreservice.repository.MenuRepository;
+import jakarta.transaction.Transactional;
+import lombok.RequiredArgsConstructor;
+import org.mapstruct.factory.Mappers;
+import org.springframework.stereotype.Service;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+@Service
+@RequiredArgsConstructor
+public class MenuService {
+
+    private final MenuRepository menuRepository;
+    private final MenuMapper menuMapper = Mappers.getMapper(MenuMapper.class);
+
+    private MenuNotFoundException throwMenuNotFoundException(UUID uuid) {
+        return new MenuNotFoundException("Menu with uuid " + uuid + " is not found");
+    }
+
+    public UUID addMenu(MenuDto menuDto) {
+        Optional.of(menuDto.getName())
+                .filter(menuRepository::existsByName)
+                .ifPresent(name -> {
+                    throw new IllegalStateException("Menu " + name + " is available");
+                });
+
+        Menu menuEntity = menuMapper.fromMenuDtoToMenuEntity(menuDto);
+        menuEntity.setUuid(UUID.randomUUID());
+        Menu savedMenu = menuRepository.save(menuEntity);
+
+        MenuDto savedMenuDto = menuMapper.fromMenuEntityToMenuDto(savedMenu);
+        return savedMenuDto.getUuid();
+    }
+
+    public List<MenuDto> findAllMenu() {
+        return menuRepository.findAll().stream()
+                .map(menuMapper::fromMenuEntityToMenuDto)
+                .toList();
+    }
+
+    public MenuDto findByMenuUuid(UUID uuid) {
+        Optional<Menu> menuByUuid = menuRepository.getMenuByUuid(uuid);
```

Comment:
1 - let's rename this var to 'menuFromDb', 'foundMenu' etc
2 - move logic of checking Optional (orElseThrow) higher

Reply:
fixed these comments